drivers/clock_control/nrf5_power_clock.c

```diff
@@ -352,3 +394,73 @@ DEVICE_AND_API_INIT(clock_nrf5_k32src,
 		    _clock_control_init, NULL, NULL, PRE_KERNEL_1,
 		    CONFIG_KERNEL_INIT_PRIORITY_DEVICE,
 		    &_k32src_clock_control_api);
+
+#if defined(CONFIG_USB) && defined(CONFIG_SOC_NRF52840)
+static void nrf5_power_usbd_power_int_enable(bool enable)
+{
+	u32_t mask;
+
+	mask = NRF_POWER_INT_USBDETECTED_MASK |
+	       NRF_POWER_INT_USBREMOVED_MASK |
+	       NRF_POWER_INT_USBPWRRDY_MASK;
+
+	if (enable) {
+		NRF_POWER->INTENSET |= mask;
+	} else {
+		NRF_POWER->INTENSET &= ~mask;
+	}
+}
+
+static bool nrf5_power_usbregstatus_vbusdet_get(void)
+{
+	return nrf_power_usbregstatus_vbusdet_get();
+}
+
+static bool nrf5_power_usbregstatus_outrdy_get(void)
+{
+	return nrf_power_usbregstatus_outrdy_get();
```

Comment:
What is this nrf_power namespace?

Reply:
Those are the HAL functions from ext. I'm using them instead of re-implementing again.